### PR TITLE
chore: Fix errors reported by clippy beta

### PR DESCRIPTION
**Pull request by kjarosh**
This fixes errors like

    error: the following explicit lifetimes could be elided: 'a
    error: empty line after doc comment
    error: unneeded `return` statement

Plus others related to doc comments.

### CODE REVIEW

**Comment by torokati44** [comment]
There are a few more with `'gc`, and there's also a `clippy::empty-line-after-doc-comments`.

**Comment by kjarosh** [comment]
Whoops, I forgot to push \;)

**Comment by torokati44** [comment]
Another one... :)

**Comment by torokati44** [comment]
They just won't run out... 😅

**Comment by torokati44** [comment]
About the (current) second commit: https://github.com/rust-lang/rust-clippy/issues/12917 may be related.

**Comment by kjarosh** [comment]
> About the (current) second commit: [rust-lang/rust-clippy#12917](https://github.com/rust-lang/rust-clippy/issues/12917) may be related.

To be honest, we should have just commented out that doc comment. It does not apply to anything as the original code has been commented out too.

**Comment by torokati44** [comment]
Yeah, but nobody suggested it until now, and even when I saw at first what you did now, it struck me as really weird, but given that "doc comments" are almost like proper language elements in Rust, it does make some weird sense to "actually comment them out", now that I think about it...

**Comment by torokati44** [comment]
_Holy mother of wide-reaching nit diffs, Crabman!_

File: core/src/bitmap/turbulence.rs [hunk 1]
```diff
@@ -38,9 +39,11 @@ fn random(seed: i64) -> i64 {
...
 #[derive(Copy, Clone)]
 struct StitchInfo {
-    width: i32, // How much to subtract to wrap for stitching.
+    /// How much to subtract to wrap for stitching.
+    width: i32,
     height: i32,
-    wrap_x: i32, // Minimum value to wrap.
+    /// Minimum value to wrap.
+    wrap_x: i32,
     wrap_y: i32,
```
Comment (torokati44) [inline core/src/bitmap/turbulence.rs:47]: Isn't this going to be weird that the doc comments are only for the first half of each pair? :thinking:

Comment (kjarosh) [reply core/src/bitmap/turbulence.rs:47]: Oh, I didn't notice those comment referred to more than one field, TBH it was weird before too... :/

I can move those comments to the doc comment of the struct or add x/y width/height variants (or something else perhaps?), which one do you prefer?

Comment (kjarosh) [reply core/src/bitmap/turbulence.rs]: Done

Comment (torokati44) [reply core/src/bitmap/turbulence.rs]: Thanks! ^^

**Review by torokati44 — APPROVED** [review]
LGTM, thank you! :tada:
I guess this is better merged sooner rather than later, so it doesn't start bitrotting on us...

**Comment by torokati44** [comment]
Oh, sorry, there's a missing space in one of the commit messages: ` out,othwerise` :P

**Comment by torokati44** [comment]
That stray `EditText` comment could also be a merge conflict resolution artifact BTW.

**Pull request merged**